regtests/bin/matrix_divider_p.sh

```diff
@@ -0,0 +1,123 @@
+# --------------------------------------------------------------------------- #
+cp matrix matrix.tmp
+HOME=${PWD%/*}
+
+maxlist1=48
+maxlist2=92
+
+#Put the job requirement/spec in "before"
+sed -e "/run_test/,\$d" matrix.tmp > before
+#Put the list of tests in "list"
+command egrep 'ww3_tp2.14|ww3_tp2.17' matrix.tmp | cat >> list_heavy
+awk '!/ww3_tp2.14/' matrix.tmp > tmpfile && mv tmpfile matrix.tmp
+awk '!/ww3_tp2.17/' matrix.tmp > tmpfile && mv tmpfile matrix.tmp
+command egrep 'mpirun|mpiexec|MPI_LAUNCH' matrix.tmp | cat >> list_mpi
+awk '!/mpirun|mpiexec|MPI_LAUNCH/' matrix.tmp > tmpfile && mv tmpfile matrix.tmp
+split -dl $maxlist1 list_mpi list_mpi_
+rm list_mpi
+matrixno1=$(ls list_mpi_* | wc -l)
+echo "Total nummber of matrix with parallel tests = $(($matrixno1 + 1)); each includes $maxlist1 tests"
+command egrep 'run_test' matrix.tmp | cat >> list_serial
+split -dl $maxlist2 list_serial list_serial_
+rm list_serial
+matrixno2=$(ls list_serial_* | wc -l)
+echo "Total nummber of matrix with serial test = $matrixno2; each includes $maxlist2 tests"
+rm matrix.tmp
+
+# --------------------------------------------------------------------------- #
+# 2.  Divide and dump in subsets                                              #
+# --------------------------------------------------------------------------- #
+# parallel jobs
+count=0
+  for i in `seq -f '%02g' 0 "$((matrixno1 - 1))"`; do
+#echo $i
+#Replace matrix.out > matrix?.out, model > model?
+  (( count = count + 1 ))
+  if [ -d "${HOME}/model${count}" ]; then rm -Rf ${HOME}/model${count}; fi
+  if [ -f "matrix${count}" ]; then rm -f matrix${count}; fi
+  cat before >> matrix$count
+  cat list_mpi_$i >> matrix$count
+  cp -r ${HOME}/model ${HOME}/model$count
+  sed -i 's/'matrix.out'/'matrix${count}.out'/gI' matrix$count
+  sed -i 's/'model'/'model${count}'/gI' matrix$count
+  echo "  echo ' '"                                                                     >> matrix$count
+  echo "  echo '     **************************************************************'"   >> matrix$count
+  echo "  echo '     *  end of WAVEWATCH III matrix$count of regression tests     *'"   >> matrix$count
+  echo "  echo '     **************************************************************'"   >> matrix$count
+  echo "  echo ' '"                                                                     >> matrix$count
+  echo "rm -r ${HOME}/model${count}"                                                    >> matrix$count
+  echo " matrix$count prepared"
+ done
+
+#serial jobs
+ for i in `seq -f '%02g' 0 "$((matrixno2 - 1))"`; do
+#echo $i
+#Replace matrix.out > matrix?.out, model > model?
+  (( count = count + 1 ))
+  if [ -d "${HOME}/model${count}" ]; then rm -Rf ${HOME}/model${count}; fi
+  if [ -f "matrix${count}" ]; then rm -f matrix${count}; fi
+  cat before >> matrix$count
+  cat list_serial_$i >> matrix$count
+  cp -r ${HOME}/model ${HOME}/model$count
```

Comment:
This line should be added to the matrix$count script.  If it is not, you cannot re-run matrix$count as you remove model$count within that scripts (see line 110).

Reply:
@JessicaMeixner-NOAA Done

Reply:
I just ran the script, and am getting a little bit of weird output: 

```
if [ -d ../model77 ]; then rm -Rf ../model77; fi
cp -r ../model7 ../model77 
...
./bin/run_test -b slurm -c hera.intel -S -T -s PR2_UQ -w work_PR2_UQ_g -m grdset_g ../model7 mww3_test_04
...
rm -r /scratch2/NCEPDEV/climate/Jessica.Meixner/PR_WW3/PR328/WW3/model7

```

Shouldn't it be something like: 
```
if [ -d ../model7 ]; then rm -rf ../model7; fi
cp -r ../model ../model7 
...
./bin/run_test -b slurm -c hera.intel -S -T -s PR2_UQ -w work_PR2_UQ_g -m grdset_g ../model7 mww3_test_04
...
rm -rf /scratch2/NCEPDEV/climate/Jessica.Meixner/PR_WW3/PR328/WW3/model7

```

Reply:
@JessicaMeixner-NOAA fixed